chia/rpc/wallet_rpc_api.py

```diff
@@ -1407,34 +1408,22 @@ async def extend_derivation_index(self, request: Dict[str, Any]) -> Dict[str, An
 
         return {"success": True, "index": updated_index}
 
-    async def get_notifications(self, request: Dict[str, Any]) -> EndpointResult:
-        ids: Optional[List[str]] = request.get("ids", None)
-        start: Optional[int] = request.get("start", None)
-        end: Optional[int] = request.get("end", None)
-        if ids is None:
+    @marshal
+    async def get_notifications(self, request: GetNotifications) -> GetNotificationsResponse:
+        if request.ids is None:
```

Comment:
This is not a thought to be addressed now.  Feel free to close this conversation after reading it.

Silently ignoring either `ids` or `start` and `end` when the other is specified is bound to cause someone some debugging hassle.  More complete schema definition etc that would make them exclusive, or actually handling both when both are requested, would be nice.  Mostly this is likely a general though.  Maybe even separate endpoints would be more appropriate.